signer/src/bin/main.rs

```diff
@@ -0,0 +1,132 @@
+
+        /// Mnemonic for account import
+        mnemonic: String,
+
+        /// File for account secrets to be written to
+        #[clap(short, long)]
+        output: String,
+    },
+
+    // Implement shared signer commands
+    #[command(flatten)]
+    Signer(Operations),
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+struct AccountSecrets {
+    mnemonic: String,
+}
+
+fn main() -> anyhow::Result<()> {
+    // Parse command line arguments
+    let args = Args::parse();
+
+    // Run commands
+    match &args.action {
+        Actions::Create { output } | Actions::Import { output, .. } => {
+            // Generate or parse mnemonic
+            let mnemonic = match &args.action {
+                Actions::Import { mnemonic, .. } => {
+                    Mnemonic::from_phrase(mnemonic, Language::English).unwrap()
+                }
+                _ => Mnemonic::new(MnemonicType::Words24, Language::English),
+            };
+
+            // Generate secrets object
+            let s = AccountSecrets {
+                mnemonic: mnemonic.to_string(),
+            };
+
+            // Check we're not overwriting an existing secret file
+            if Path::new(output).exists() {
+                return Err(anyhow::anyhow!(
+                    "creation would overwrite existing secrets file '{}'",
+                    output
+                ));
+            }
+
+            // Otherwise write out new secrets
+            write_output(output, &s)?;
+
+            info!("Account secrets written to '{}'", output);
+        }
+        Actions::Signer(c) => {
+            // Load account secrets
+            let secrets: AccountSecrets = read_input(&args.secret_file)?;
+            let mnemonic = Mnemonic::from_phrase(&secrets.mnemonic, Language::English)?;
+
+            // Perform SLIP-0010 derivation
+            let account_index = c.account_index();
+            let slip10key = mnemonic.derive_slip10_key(account_index);
+
+            // Generate account from secrets
+            let a = Account::from(&slip10key);
+
+            debug!("Using account: {:?}", a);
+
+            // Handle standard commands
+            match c {
+                Operations::GetAccount { output, .. } => {
+                    Operations::get_account(&a, account_index, output)?
+                }
+                Operations::SyncTxos { input, output, .. } => {
+                    Operations::sync_txos(&a, input, output)?
+                }
+                Operations::SignTx { input, output, .. } => {
+                    // Setup local ring signer
+                    let ring_signer = LocalRingSigner::from(&AccountKey::new(
+                        a.spend_private_key().as_ref(),
+                        a.view_private_key().as_ref(),
+                    ));
+
+                    // Perform transaction signing
+                    Operations::sign_tx(&ring_signer, input, output)?;
+                }
+                _ => (),
```

Comment:
This is fishy to me, I think if there are any operations the tool doesn't support it should panic when someone tries to use them

Reply:
Does this update seem sufficient?